File: app/src/main/java/com/example/dcct/presenter/QueryPresenter.java
package com.example.dcct.presenter;

import com.example.dcct.base.BasePresenter;
import com.example.dcct.bean.BackResultData;
import com.example.dcct.bean.PostQueryEntity;
import com.example.dcct.bean.QueryResultEntity;
import com.example.dcct.model.Impl.QueryModelImp;
import com.example.dcct.model.QueryModel;
import com.example.dcct.view.QueryCallback;

import java.util.List;


public class QueryPresenter extends BasePresenter {
    private QueryModel mQueryModel = new QueryModelImp();

    public void fetchQueryResult(PostQueryEntity postQueryEntity){
        if (mQueryModel != null && mCallBackView.get() != null) {
            mQueryModel.postQueryInformation( postQueryEntity, new QueryModel.ReturnQueryResult() {
                @Override
                public void onComplete(BackResultData<List<QueryResultEntity>> listBackResultData) {
                    ((QueryCallback)mCallBackView.get()).onLoadQueryData( listBackResultData );
                }

                @Override
                public void onError(String msg) {
                    ((QueryCallback)mCallBackView.get()).showErrorMsg( msg );
                }
            } );
        }
    }
}

File: app/src/main/java/com/example/dcct/presenter/QueryPresenter.kt
package com.example.dcct.presenter

import com.example.dcct.base.BasePresenter
import com.example.dcct.bean.BackResultData
import com.example.dcct.bean.PostQueryEntity
import com.example.dcct.bean.QueryResultEntity
import com.example.dcct.model.impl.QueryModelImp
import com.example.dcct.model.QueryModel
import com.example.dcct.model.QueryModel.ReturnQueryResult
import com.example.dcct.view.QueryCallback

class QueryPresenter : BasePresenter<QueryCallback>() {
    private val mQueryModel: QueryModel = QueryModelImp()
    fun fetchQueryResult(postQueryEntity: PostQueryEntity) {
        if (mCallBackView!!.get() != null) {
            mQueryModel.postQueryInformation(postQueryEntity, object : ReturnQueryResult {
                override fun onComplete(listBackResultData: BackResultData<List<QueryResultEntity>>) {
                    mCallBackView!!.get()!!.onLoadQueryData(listBackResultData)
                }

                override fun onError(msg: String) {
                    mCallBackView!!.get()!!.showErrorMsg(msg)
                }
            })
        }
    }
}